rust/otap-dataflow/crates/otap/src/otap_batch_processor.rs

```diff
@@ -0,0 +1,922 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! OTAP batch processor (skeleton)
+//!
+//! Mirrors the configuration shape of the OpenTelemetry Collector batchprocessor,
+//! but operates on OtapPdata. This integration is aligned with the OtapPdata combine work.
+//!
+//! Current scope:
+//! - Buffers incoming messages and flushes by count or timer.
+//! - Does not directly mutate Arrow RecordBatches; merging/splitting is delegated to upstream OTAP
+//!   batching utilities.
+//! - Does not directly split Arrow batches; any chunking is handled by upstream batching and emitted
+//!   as multiple outputs when necessary.
+//! - Partitions emission by signal type (logs/metrics/traces) when possible, but does not re-order
+//!   or sort by time.
+//!
+//! Merging, splitting, and group-key logic are delegated to OtapPdata helpers and upstream batching
+//! utilities in otel-arrow-rust.
+
+use crate::OTAP_PROCESSOR_FACTORIES;
+use crate::pdata::OtapPdata;
+use async_trait::async_trait;
+use linkme::distributed_slice;
+use otap_df_config::error::Error as ConfigError;
+use otap_df_config::experimental::SignalType;
+use otap_df_config::node::NodeUserConfig;
+use otap_df_engine::config::ProcessorConfig;
+use otap_df_engine::error::Error as EngineError;
+use otap_df_engine::local::processor as local;
+use otap_df_engine::message::Message;
+use otap_df_engine::node::NodeId;
+use otap_df_engine::processor::ProcessorWrapper;
+use serde::Deserialize;
+use serde_json::Value;
+use std::num::NonZeroU64;
+use std::sync::Arc;
+// For optional conversion during flush/partitioning
+use otel_arrow_rust::otap::OtapArrowRecords;
+use otel_arrow_rust::otap::batching::make_output_batches as low_make_output_batches;
```

Comment:
nit: the low_ prefix is a bit confusing :-)

Reply:
import name was fixed